src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,16 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String name;
+    private String number;
+    private String bodyType;
+    private String engineType;
+    private Engine engine;
 
+    public Car() {
+    }
+
+    public Car clone() throws CloneNotSupportedException {
+        return (Car)super.clone();
```

Comment:
that would work for shallow cloning, but the thing is that we need deep cloning cause we have Engine object with his own fields as an Car's field

Reply:
changed method clone() in Car class

Reply:
added another commit with new constructors and rewritten clone method